src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -408,87 +436,38 @@ private int FillBuffer(int minSize)
                         throw new IOException(SR.net_io_eof);
                     }
 
-                    return 0;
+                    return new ValueTask<int>(0);
                 }
 
                 _internalBufferCount += bytes;
             } while (_internalBufferCount < minSize);
 
-            return minSize;
-        }
+            return new ValueTask<int>(minSize);
 
-        private static void CompleteFromCompletedTask(Task<int> task, AsyncProtocolRequest asyncRequest)
-        {
-            Debug.Assert(task.IsCompleted);
-            if (task.IsCompletedSuccessfully)
+            async Task<int> InternalFillBufferAsync(TReadAdapter adap, ValueTask<int> task, int min, int initial)
```

Comment:
private

Reply:
It'll probably be slightly cheaper if this takes a `Task<int>` instead of a `ValueTask<int>`.  At the call site we already checked for `IsCompletedSuccessfully`, so we're only calling this function if either it completed due to an error (in which case perf isn't a big issue) or due to an asynchronous completion, so if the call site uses `.AsTask()` and then this takes a `Task<int>`, we'll save an `int` field in the state machine.

Reply:
Can't be private is local function?

Reply:
> Can't be private is local function?

Right.  Ignore that comment.